src/scaladoc/scala/tools/nsc/doc/html/resource/lib/index.js

```diff
@@ -532,6 +532,7 @@ function searchAll() {
     scheduler.clear("search"); // clear previous search
     maxJobs = 1; // clear previous max
     var searchStr = $("#textfilter input").attr("value").trim() || '';
+    searchStr = escape(searchStr);
```

Comment:
What about using `encodeURIComponent` instead? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/escape

Reply:
I don't think this is the correct solution regardless - as pointed about by Harrison, it prevents you from searching for some things (like `:+`). The problem is that the untrusted string is embedded in a tag creation below.